openmls/src/treesync/node/leaf_node.rs

```diff
@@ -410,20 +432,47 @@ impl TlsDeserializeTrait for LeafNodeTbs {
 /// } LeafNode;
 /// ```
 #[derive(
-    Debug, Clone, PartialEq, Eq, Serialize, Deserialize, TlsSerialize, TlsDeserialize, TlsSize,
+    Debug, Clone, PartialEq, Eq, Serialize, Deserialize, TlsSize, TlsSerialize, TlsDeserialize,
 )]
-pub struct LeafNode {
+pub struct LeafNode<State> {
     payload: LeafNodePayload,
     signature: Signature,
+    phantom: PhantomData<State>,
 }
 
-impl From<OpenMlsLeafNode> for LeafNode {
+// -------------------------------------------------------------------------------------------------
+
+impl From<OpenMlsLeafNode> for LeafNode<StateUnvalidated> {
     fn from(leaf: OpenMlsLeafNode) -> Self {
-        leaf.leaf_node
+        LeafNode {
+            payload: leaf.leaf_node.payload,
+            signature: leaf.leaf_node.signature,
+            phantom: Default::default(),
+        }
+    }
+}
+
+impl From<LeafNode<StateKeyPackage>> for LeafNode<StateValidated> {
+    fn from(value: LeafNode<StateKeyPackage>) -> Self {
+        LeafNode {
+            payload: value.payload,
+            signature: value.signature,
+            phantom: Default::default(),
+        }
+    }
+}
+
+impl From<LeafNode<StateAddUpdateCommit>> for LeafNode<StateValidated> {
+    fn from(value: LeafNode<StateAddUpdateCommit>) -> Self {
+        LeafNode {
+            payload: value.payload,
+            signature: value.signature,
+            phantom: Default::default(),
+        }
     }
 }
 
-impl LeafNode {
+impl<T> LeafNode<T> {
```

Comment:
This was an easy way to make the code compile again. Later, we do want to have separate `impl`s for every typestate.